src/OrchardCore/OrchardCore.DisplayManagement/Descriptors/Interfaces.cs

```diff
@@ -1,5 +1,6 @@
 using System.Threading.Tasks;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
```

Comment:
Any perf degradation?

Reply:
If `AddShapeAttributes<T>()` is called once per shape type it is as before.

If called multiple times for a given shape type, less registrations, so better for perf.

Reply:
I meant, does the reflection here makes it slower on an equivalent request?

Reply:
There is no additional reflection, just a generic extension helper as before.